### PR TITLE
Setup providers

### PR DESCRIPTION
**Pull request by leongold**

### CODE REVIEW

**Comment by leongold** [comment]
@SchSeba, multus env fails with:

Dependency Installed:
  libpcap.x86_64 14:1.5.3-11.el7                                                

Complete!
**Job for openvswitch.service canceled.** <--- due to this I guess?
make: *** [cluster-up] Error 1
+ release_download_lock /var/lib/stdci/shared/download_rhel_image.lock
+ [[ -e /var/lib/stdci/shared/download_rhel_image.lock ]]
+ release_download_lock
+ [[ -e '' ]]
+ make cluster-down
./cluster/down.sh
Took 804 seconds
====================

**Comment by leongold** [comment]
full log: [](https://jenkins.ovirt.org/job/kubevirt_ovs-cni_standard-check-pr/81//artifact/check-patch.openshift-3.10-multus-release.el7.x86_64/mock_logs/script/stdout_stderr.log)

**Comment by SchSeba** [comment]
@leongold Can you please try to use `fa6d763af28634417200ee35e095cc56d8ad4039e58f70c3a04a59d498011ec5` as the image sha for the openshift provider?

`cluster/os-3.10.0-multus/provider.sh`

**Comment by leongold** [comment]
> @leongold Can you please try to use `fa6d763af28634417200ee35e095cc56d8ad4039e58f70c3a04a59d498011ec5` as the image sha for the openshift provider?
> 
> `cluster/os-3.10.0-multus/provider.sh`

Cool, works! What's the relevant difference?

**Comment by SchSeba** [comment]
This is the last image for the provider. It always important to use the same one we have on kubevirt.

**Comment by leongold** [comment]
Great. I'll just add kubevirt deployment to this PR, next would be OVS with a minimal test?

**Comment by SchSeba** [comment]
@leongold Please remove the for loop inside the up.sh command, and change the digest of the k8s provider to `5d57cacc0d845b7ee20722e3481a1e9525336dd3ee8ac479037bce531d48c9f8`.

**Comment by leongold** [comment]
@SchSeba, great, works, we can merge this and move to OVS?

File: automation/test.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+kubectl get nodes
+
+make cluster-sync
+
+if [[ $TARGET =~ openshift.* ]]; then
+  kubectl adm policy add-scc-to-user privileged system:serviceaccount:kube-system:kubevirt-privileged
+  kubectl adm policy add-scc-to-user privileged system:serviceaccount:kube-system:kubevirt-controller
+  kubectl adm policy add-scc-to-user privileged system:serviceaccount:kube-system:kubevirt-apiserver
+fi
+kubectl apply -f https://github.com/kubevirt/kubevirt/releases/download/$KUBEVIRT_VERSION/kubevirt.yaml
```
[the remark this answers is not in the document]
Comment (SchSeba) [reply automation/test.sh]: Deploy upstream kubervirt to test the ovs-cni plugin with virtual machines (sounds good?)

[the remark this answers is not in the document]
Comment (leongold) [reply automation/test.sh]: Done

File: automation/test.sh [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    current_time=$((current_time + sample))
+    if [ $current_time -gt $timeout ]; then
+      exit 1
+    fi
+  done
+  kubectl get pods -n $i
+done
+
+kubectl version
```
[the remark this answers is not in the document]
Comment (leongold) [reply automation/test.sh]: Done

File: automation/test.sh [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    current_time=$((current_time + sample))
+    if [ $current_time -gt $timeout ]; then
+      exit 1
+    fi
+  done
+
+  # Make sure all containers are ready
+  current_time=0
+  while [ -n "$(kubectl get pods -n $i -o'custom-columns=status:status.containerStatuses[*].ready' --no-headers | grep false)" ]; do
```
[the remark this answers is not in the document]
Comment (leongold) [reply automation/test.sh]: Done

File: automation/test.sh [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+timeout=300
+sample=30
+
+for i in ${namespaces[@]}; do
+  # Wait until kubevirt pods are running
+  current_time=0
+  while [ -n "$(kubectl get pods -n $i --no-headers | grep -v Running)" ]; do
+    echo "Waiting for kubevirt pods to enter the Running state ..."
+    kubectl get pods -n $i --no-headers | >&2 grep -v Running || true
+    kubectl describe pods -n kube-system
```
[the remark this answers is not in the document]
Comment (leongold) [reply automation/test.sh]: Done

File: automation/test.sh [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    kubectl get nodes --no-headers
+    kubectl_rc=$?
+    sleep 10
+done
+set -e
+
+echo "Nodes are ready:"
+kubectl get nodes
+
+make cluster-sync
```
[the remark this answers is not in the document]
Comment (leongold) [reply automation/test.sh]: Done

**Pull request merged**